libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts

```diff
@@ -588,6 +594,45 @@ export class WorkspaceInstallManager extends BaseInstallManager {
       );
     }
 
+    // Hoist any dependencies for this subspace if splitWorkspaceCompatibility is enabled
+    if (
+      this.rushConfiguration.subspacesFeatureEnabled &&
+      this.rushConfiguration.subspacesConfiguration?.splitWorkspaceCompatibility &&
+      subspace.subspaceName.startsWith('split_') &&
+      subspace.getProjects().length === 1
+    ) {
+      // Find the .modules.yaml file in the subspace temp/node_modules folder
+      const tempNodeModulesPath: string = `${subspace.getSubspaceTempFolder()}/node_modules`;
+      const modulesFilePath: string = `${tempNodeModulesPath}/${RushConstants.pnpmModulesFilename}`;
+
+      const modulesContent: string = FileSystem.readFile(modulesFilePath);
+      const yamlContent: IPnpmModules = yaml.load(modulesContent, { filename: modulesFilePath });
+      const { hoistedDependencies } = yamlContent;
+      const subspaceProject: RushConfigurationProject = subspace.getProjects()[0];
+      const projectNodeModulesPath: string = `${subspaceProject.projectFolder}/node_modules`;
+      for (const value of Object.values(hoistedDependencies)) {
+        for (const [filePath, type] of Object.entries(value)) {
+          if (type === 'public') {
+            // If we don't already have a symlink for this package, create one
+            if (!fs.existsSync(`${projectNodeModulesPath}/${filePath}`)) {
+              // Ensure origin folder exists
+              const parentDirSep: string[] = `${projectNodeModulesPath}/${filePath}`.split(path.sep);
+              parentDirSep.pop();
+              const parentDir: string = parentDirSep.join(path.sep);
+              if (!fs.existsSync(parentDir)) {
+                fs.mkdirSync(parentDir, { recursive: true });
```

Comment:
This can be done more efficiently using the `async` `FileSystem.ensureFolderAsync()`

Reply:
Changed